changelogs/unreleased/gh-10538-add-on-box-status-callback-to-roles.md

```diff
@@ -0,0 +1,3 @@
+## feature/config
+
+* Introduced `on_box_status` callback for roles (gh-10538).
```

Comment:
We agreed that it would look more accurate to call it `on_event` and pass either `box.status` or something about config (`'config.apply'`?[^1]) into the `key` argument.

I still think that we should pass role's config as the first argument, so the same function can be assigned to `apply` and `on_event`.

[^1]: We have `config.info` event, but it is called at slightly different point. Maybe it is not a god idea to mix them.

Reply:
Renamed it to `on_event`, `key = 'box.status` if the callback was triggered by `box.status` system event, and `key = 'config.apply'` if it was triggered because of configuration update.